src/entities/person/possible-duplicates.ts

```diff
@@ -0,0 +1,82 @@
+import { Universe } from 'src/universe'
+import { EntityRawPayload, UniverseEntityOptions } from '../_base'
+import { PersonRawPayload, PersonPhonenumberRawPayload } from './person'
+import { ChannelUserRawPayload } from './channel-user'
+import { BaseError } from '../../errors'
+
+export interface PossibleDuplicatesRawPayload extends EntityRawPayload{
+  readonly person_id: string
+  readonly strategies: PossibleDuplicatesStrategies
+}
+
+interface PossibleDuplicatesStrategies {
+  global_phonenumber?: StrategiesResult[]
+  name_exact_matches?: StrategiesResult[]
+}
+
+interface StrategiesResult {
+  document: StrategiesDocument
+  kind: string
+}
+
+interface StrategiesDocument {
+  person: Pick<PersonRawPayload, 'id'| 'kind' | 'name'>
+  channel_user?: Pick<ChannelUserRawPayload, 'id' | 'name' | 'phone'>
+  phone_number?: Pick<PersonPhonenumberRawPayload, 'id' | 'value'>
+}
+
+interface PossibleDuplicationOptions extends UniverseEntityOptions {
+  rawPayload: PossibleDuplicatesRawPayload
+}
+
+export class PossibleDuplication {
+  protected universe: Universe
+  protected apiCarrier: Universe
+  protected http: Universe['http']
+  protected options: PossibleDuplicationOptions
+  public initialized: boolean
+  public endpoint: string
+
+  public id?: string
+  public personId: PossibleDuplicatesRawPayload['person_id']
+  public strategies: PossibleDuplicatesRawPayload['strategies']
+
+  constructor (options: PossibleDuplicationOptions) {
+    this.universe = options.universe
+    this.apiCarrier = options.universe
+    this.http = options.http
+    this.options = options
+    this.initialized = options.initialized ?? false
+    this.endpoint = ''
```

Comment:
We seem to never set the endpoint anywhere anymore. How does it work then? 😅

Reply:
we do actually, but in `people.checkDuplicates()` instead and here is my thinking why and please comment and correct me if I'm wrong:
- we cannot set the endpoint in the constructor of `PossibleDuplication` because it will be instantiate after the call when the endpoint returns it, so the moment we call `{{HOST}}/api/v0/people/:id/check/duplicates` it won't be there yet
- but `people` will so `this.endpoint` will be what is set in its constructor (`this.endpoint = 'api/v0/people'`)
- we complete the endpoint then in the aforementioned `people.checkDuplicates()`
- I thought this could be why after Nick showed me how to test our endpoint in the `agent-ui/src/pages/dev/Sandbox.vue/mounted()::his.$universe.person({ id: '2c26b6f1-3a45-4b94-ac03-553d8e7ed832' }).checkDuplicates().then(res => { console.log(res) })` <- you can see how we pass the person id to instantiate the person we call the `checkDuplicates()` method for
WDYT?

Reply:
This would become unnecessary if my last comment applies.